Backpack/Classes/Gradient/BPKGradient.m

```diff
@@ -0,0 +1,111 @@
+#import "BPKGradient.h"
+
+#import <Backpack/Color.h>
+#import <UIKit/UIKit.h>
+
+CGPoint startPointForDirection(BPKGradientDirection direction){
+    switch (direction) {
+        case BPKGradientDirectionUp:
+            return CGPointMake(0.5, 1.0);
+        case BPKGradientDirectionDown:
+            return CGPointMake(0.5, 0.0);
+        case BPKGradientDirectionLeft:
+            return CGPointMake(1.0, 0.5);
+        case BPKGradientDirectionRight:
+            return CGPointMake(0.0, 0.5);
+        case BPKGradientDirectionTopLeft:
+            return CGPointMake(1.0, 1.0);
+        case BPKGradientDirectionTopRight:
+            return CGPointMake(0.0, 1.0);
+        case BPKGradientDirectionBottomLeft:
+            return CGPointMake(1.0, 0.0);
+        case BPKGradientDirectionBottomRight:
+            return CGPointMake(0.0, 0.0);
+        default:
+            return CGPointMake(0.0, 0.0);
+            break;
+    }
+}
+
+CGPoint endPointForDirection(BPKGradientDirection direction){
+    switch (direction) {
+        case BPKGradientDirectionUp:
+            return CGPointMake(0.5, 0.0);
+        case BPKGradientDirectionDown:
+            return CGPointMake(0.5, 1.0);
+        case BPKGradientDirectionLeft:
+            return CGPointMake(0.0, 0.5);
+        case BPKGradientDirectionRight:
+            return CGPointMake(1.0, 0.5);
+        case BPKGradientDirectionTopLeft:
+            return CGPointMake(0.0, 0.0);
+        case BPKGradientDirectionTopRight:
+            return CGPointMake(1.0, 0.0);
+        case BPKGradientDirectionBottomLeft:
+            return CGPointMake(0.0, 1.0);
+        case BPKGradientDirectionBottomRight:
+            return CGPointMake(1.0, 1.0);
+        default:
+            return CGPointMake(1.0, 1.0);
+            break;
+    }
+}
+
+@interface BPKGradient()
```

Comment:
We should include a `NS_ASSUME_NONNULL_BEGIN`/`NS_ASSUME_NONNULL_END` around both this interface and the implementation too